map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -0,0 +1,254 @@
+package edu.artic.map
+
+import android.graphics.Bitmap
+import android.os.Bundle
+import android.view.View
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.target.SimpleTarget
+import com.bumptech.glide.request.transition.Transition
+import com.fuzz.rx.disposedBy
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.MapView
+import com.google.android.gms.maps.MapsInitializer
+import com.google.android.gms.maps.model.*
+import edu.artic.analytics.ScreenCategoryName
+import edu.artic.db.models.*
+import edu.artic.map.helpers.toLatLng
+import edu.artic.map.util.ArticObjectMarkerGenerator
+import edu.artic.map.util.DepartmentMarkerGenerator
+import edu.artic.map.util.GalleryNumberMarkerGenerator
+import edu.artic.viewmodel.BaseViewModelFragment
+import io.reactivex.android.schedulers.AndroidSchedulers
+import kotlinx.android.synthetic.main.fragment_map.*
+import kotlin.reflect.KClass
+
+class MapFragment : BaseViewModelFragment<MapViewModel>() {
+
+    override val viewModelClass: KClass<MapViewModel>
+        get() = MapViewModel::class
+    override val title: String
+        get() = ""
+    override val layoutResId: Int
+        get() = R.layout.fragment_map
+    override val screenCategory: ScreenCategoryName
+        get() = ScreenCategoryName.Map
+    // This class is due to the fact that kotlinx will return null in ondestroy if used to get
+    // the proper view
+    private lateinit var destroyableMapView: MapView
+    lateinit var map: GoogleMap
+
+    val currentMarkers = mutableListOf<Marker>()
+
+    lateinit var objectMarkerGenerator: ArticObjectMarkerGenerator
+    lateinit var galleryNumberGenerator: GalleryNumberMarkerGenerator
+    lateinit var departmentMarkerGenerator: DepartmentMarkerGenerator
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        objectMarkerGenerator = ArticObjectMarkerGenerator(view.context)
+        galleryNumberGenerator = GalleryNumberMarkerGenerator(view.context)
+        departmentMarkerGenerator = DepartmentMarkerGenerator(view.context)
+
+        destroyableMapView = mapView
+        mapView.onCreate(savedInstanceState)
+        MapsInitializer.initialize(view.context)
+        mapView.getMapAsync { map ->
+            this.map = map
+            map.setMinZoomPreference(18f)
+            map.setMaxZoomPreference(21f)
+            map.setLatLngBoundsForCameraTarget(
+                    LatLngBounds(
+
+                            LatLng(41.878523, -87.623689),
```

Comment:
Leave comment here on why this is necessary (just like this is the bounds of the museum)